### ISSUE TITLE
Cellular data option does not connect to the internet anymore

### ISSUE THREAD

**Issue by Stevie00911** [HelloVolla/ubuntu-touch-beta-tests#70]
Ubuntu 16.04 (2020-W44) release candidate:
After some days of using the phone the "Cellular data" option does not connect to the internet anymore.
I can switch the Celular data slider to on, but I do not get the "3G" symbol in the notification area.
Only after a reboot the connection to the internet can be activated again.

**Comment by JamiKettunen** [comment]
As far as I know there sadly isn't a proper/simpler way (yet) to capture Ofono logs, but could you run these to enable verbose logging and run `grep -i 'ofono' /var/log/syslog` later to get logs when you encounter this issue again (you can try toggling cellular data a few times just to be sure the issue logged)? That would greatly help with debugging this issue:
```
mount | grep -q ' / .*ro' && sudo mount -o remount,rw /
sudo sed 's/ofonod/ofonod -d/' -i /etc/init/ofono.override
sudo initctl restart ofono
```
To disable verbose logs again you can run the same commands but use the following `sed`: `sudo sed 's/ofonod -d/ofonod/' -i /etc/init/ofono.override`

**Comment by lhooge** [comment]
Hi,
my provider is congstar and I have also no internet connection when I connect over cellular data. But I did get the L symbol in the notification bar. Maybe it is related to Stevie problem.

The following ofono logs are captured:

`Nov 29 13:39:36 ubuntu-phablet ofonod[2359]: Terminating
Nov 29 13:39:36 ubuntu-phablet ofonod[2359]: Deactivating context: 1
Nov 29 13:39:36 ubuntu-phablet ofonod[2359]: SMS History remove for modem: 0x6047eb0
Nov 29 13:39:36 ubuntu-phablet ofonod[2359]: Exit
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1415] ofono: (ril_0): IPv4 static Settings:
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <warn>  [1606653576.1417] ofono: (ril_0): Settings 'Interface' missing
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1418] ofono: (ril_0): emitting PREPARE_RESULT: FALSE
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: oFono version 1.23
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1593] oFono modem removed: /ril_0
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1656] oFono modem removed: /ril_1
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1696] SettingsPlugin-Ofono: reading configuration for IMSI 262016541403240
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1702] SettingsPlugin-Ofono: 'context1' - Preferred = 'true'
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1704] SettingsPlugin-Ofono: found internet context 'Internet' (context1)
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1705] SettingsPlugin-Ofono: context 'Internet' (context1) already exported
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1708] SettingsPlugin-Ofono: reading configuration for IMSI 262016541403240
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1712] SettingsPlugin-Ofono: 'context1' - Preferred = 'true'
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1714] SettingsPlugin-Ofono: found internet context 'Internet' (context1)
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1714] SettingsPlugin-Ofono: context 'Internet' (context1) already exported
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1717] SettingsPlugin-Ofono: reading configuration for IMSI 262016541403240
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1721] SettingsPlugin-Ofono: 'context1' - Preferred = 'true'
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1722] SettingsPlugin-Ofono: found internet context 'Internet' (context1)
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1723] SettingsPlugin-Ofono: context 'Internet' (context1) already exported
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1725] SettingsPlugin-Ofono: reading configuration for IMSI 262016541403240
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1728] SettingsPlugin-Ofono: 'context1' - Preferred = 'true'
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1730] SettingsPlugin-Ofono: found internet context 'Internet' (context1)
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1730] SettingsPlugin-Ofono: context 'Internet' (context1) already exported
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1731] SettingsPluginOfono: /var/lib/ofono/262016541403240/gprs.9G9SU0 got removed.
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1732] SettingsPlugin-Ofono: reading configuration for IMSI 262016541403240
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1737] SettingsPlugin-Ofono: 'context1' - Preferred = 'true'
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1739] SettingsPlugin-Ofono: found internet context 'Internet' (context1)
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1739] SettingsPlugin-Ofono: context 'Internet' (context1) already exported
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1741] SettingsPlugin-Ofono: reading configuration for IMSI 262016541403240
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1745] SettingsPlugin-Ofono: 'context1' - Preferred = 'true'
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1747] SettingsPlugin-Ofono: found internet context 'Internet' (context1)
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.1748] SettingsPlugin-Ofono: context 'Internet' (context1) already exported
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: Excluding udev hardware detection
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: Excluding Dialup modem driver
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: Excluding External Hands-Free Profile Plugin
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: Excluding Dial-up Networking Profile Plugins
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: Excluding Provisioning Plugin
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: Excluding Smart Messaging Plugin
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: Initializing RIL binder transport plugin.
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: Register Profile interface failed: /bluetooth/profile/hfp_ag
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <warn>  [1606653576.2320] SettingsPluginOfono: unexpected event type '4'
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <warn>  [1606653576.2381] SettingsPluginOfono: unexpected event type '4'
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: parse_devices_reply: found 1st battery device: /org/freedesktop/UPower/devices/battery_cw_bat
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: slot1 version 100
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: slot2 version 100
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.3523] oFono modem appeared: /ril_1
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <warn>  [1606653576.3600] ofono (ril_1): check_connection /262016541403240/context1 failed: no IMSI
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <warn>  [1606653576.3603] ofono (ril_1): check_connection /262016541057112/context1 failed: no IMSI
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <warn>  [1606653576.3621] ofono (ril_1): check_connection /262016541403240/context1 failed: no IMSI
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <warn>  [1606653576.3623] ofono (ril_1): check_connection /262016541057112/context1 failed: no IMSI
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <info>  [1606653576.3990] oFono modem appeared: /ril_0
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <warn>  [1606653576.4090] ofono (ril_0): check_connection /262016541403240/context1 failed: no IMSI
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <warn>  [1606653576.4094] ofono (ril_0): check_connection /262016541057112/context1 failed: no IMSI
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <warn>  [1606653576.4111] ofono (ril_0): check_connection /262016541403240/context1 failed: no IMSI
Nov 29 13:39:36 ubuntu-phablet NetworkManager[2277]: <warn>  [1606653576.4114] ofono (ril_0): check_connection /262016541057112/context1 failed: no IMSI
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: SIM card OK
Nov 29 13:39:36 ubuntu-phablet ofonod[22514]: Requested file structure differs from SIM: 6fb7
Nov 29 13:39:37 ubuntu-phablet ofonod[22514]: SMS History Probe for modem: 0x119bc010
Nov 29 13:39:37 ubuntu-phablet ofonod[22514]: Registered interface org.ofono.NetworkTime, path /ril_0
Nov 29 13:39:37 ubuntu-phablet ofonod[22514]: data reg changed 0 -> 2 (searching), attached 0
Nov 29 13:39:37 ubuntu-phablet ofonod[22514]: nw selection is already auto
Nov 29 13:39:37 ubuntu-phablet ofonod[22514]: Requested file structure differs from SIM: 6fb7
Nov 29 13:39:37 ubuntu-phablet ofonod[22514]: Unable to read waiting messages numbers from SIM
Nov 29 13:39:37 ubuntu-phablet ofonod[22514]: Unable to read mailbox identifies from SIM
Nov 29 13:39:39 ubuntu-phablet ofonod[22514]: data reg changed 2 -> 1 (registered), attached 1
Nov 29 13:39:44 ubuntu-phablet ofonod[22514]: Activating context: 1
Nov 29 13:39:44 ubuntu-phablet ofonod[22514]: setting up data call
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8243] ofono: (ril_0): IPv4 static Settings:
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8248] ofono (ril_0) Address: 10.20.84.15/24
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8250] ofono: (ril_0):  Gateway: 10.20.84.15
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8252] ofono: (ril_0): DNS: 10.74.210.210
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8253] ofono: (ril_0): DNS: 10.74.210.211
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8255] ofono: (ril_0): emitting PREPARE_RESULT: TRUE
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8256] ofono: (ril_0): IPv4 static Settings:
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8277] ofono (ril_0) Address: 10.20.84.15/24
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8280] ofono: (ril_0):  Gateway: 10.20.84.15
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8281] ofono: (ril_0): DNS: 10.74.210.210
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8282] ofono: (ril_0): DNS: 10.74.210.211
Nov 29 13:39:44 ubuntu-phablet NetworkManager[2277]: <info>  [1606653584.8283] ofono: (ril_0): emitting PREPARE_RESULT: TRUE
`

**Comment by gbdomubpkm** [comment]
With last RC : For my first day of Volla use, i also have the bad surprise of NO INTERNET (no cellular data). I       tried to change the APN configurations, it doesn't help. A reboot doesn't help.
Everything works well automatically on the BQ E5 HD OTA-14,   OPO  OTA-14...
![screenshot20311030_100705360](https://user-images.githubusercontent.com/50362604/101223528-83d49300-368c-11eb-9761-eccd9cb526f1.png)
What can i do ?

**Comment by gbdomubpkm** [comment]
For info, everything works perfectly well with Volla OS, the good configuration was set up automatically.

**Comment by gbdomubpkm** [comment]
New tests and reports if it can help :
- I checked and the problem is the same with OTA-14 (stable): no internet connection
![pas données cellulaires](https://user-images.githubusercontent.com/50362604/101950440-96ebe380-3bf5-11eb-8902-ea6d90ead2c8.png)

- APN default settings are good with OTA-14
![bons reg APN 1](https://user-images.githubusercontent.com/50362604/101950600-d4507100-3bf5-11eb-86b5-747fdc7d19f5.png)

![bons reg internet](https://user-images.githubusercontent.com/50362604/101948546-8423df80-3bf2-11eb-8fe3-c03ba1ec9cf3.png)
![bons reg mms](https://user-images.githubusercontent.com/50362604/101948573-91d96500-3bf2-11eb-8426-48a95b349435.png)
(Bouygues  MMS was not enabled: I checked it).
- I don't understand why I have twice the same Bouygues Telecom/Internet configuration (I checked it), one of which is not activated and the one not activated cannot be deleted (pressing the bin has no action).
![pas effacabble](https://user-images.githubusercontent.com/50362604/101948937-2b087b80-3bf3-11eb-8ae7-d61b26b1b634.png)
- I suddenly saw the loss of SIM card recognition ('no SIM card detected' message): the SIM card is re-detected after a reboot.

NOTE
For all smartphones I own outside the Volla (BQ E5 HD, OPO), there is a single configuration for INTERNET and MMS.
![conf com](https://user-images.githubusercontent.com/50362604/101949140-6c992680-3bf3-11eb-8e72-4a77f88b5a6e.png)
![conf com2](https://user-images.githubusercontent.com/50362604/101949243-93575d00-3bf3-11eb-8718-23e6ab98ca93.png)


QUESTION
Unlike Android (base of Volla OS), in the APN settings and whatever the smartphone (Volla, BQ, OPO..), I don't see the MCC :  208,  MNC : 20  and APN type   default.supl.mms   parameters (here for Bouygues). I don't know if these parameter absences have a bad impact on Ubuntu touch, not only for me but for all UT users.

**Comment by gbdomubpkm** [comment]
To check or not the presence of the problem, this time I flashed the smartphone (in dev channel) by using the installer: everything went well, but nothing has changed: the problem is there! And I tried all the possible manipulations after in all directions: reboot, airplane mode, network change... As a bonus, I had a loss of SIM card recognition (as no more SIM card): recognized after reboot 😉

**Comment by JamiKettunen** [comment]
If someone here is encountering this issue, could you run the following in a terminal to attempt capturing logs when it happens? I've tried this once already, but restarting `ofonod` made the issue disappear again :/
```
sudo -s
stop ofono
ofonod -n -ddd 2>&1 | tee ofono.log
```
The log file can be sent to either me (@ Deathmist1337) or @NotKit on Telegram.

**Comment by gbdomubpkm** [comment]
@JamiKettunen I expect you receoved the logs.
 ofono-1.log when i tried to install a new conf, i lost the Sim (no Sim detected)., then, after a reboot, i activate mms conf, new reboot, no sim detected again.
  Ofono.log : after a new reboot,  i have internet and mms conf installed but no cellular data, no internet.
I tried with two different sim and same problem.
Sorry, i'm not a specialist neither a developper so i do what i can even for logs.. 
Expect the logs usefull to you.

**Comment by deyanyosifov** [comment]
I have some similar problems. I have two SIM cards and 4G LTE works only with the SIM card that is inserted in slot 1, no matter which. It frequently happens that when I try to initiate a call through the SIM card in slot 2, I either get a message "No SIM card selected" or it appears that the phone is dialling the number but I hear no signal at all (in both cases the network availability indicator shows full signal strength). In such cases, when I restart the phone usually the problem disappears and I am able to make a call but some time later the problem appears again. Sometimes, after a restart I get prompted to enter the PIN code only for the SIM card in slot 1 and in this case the SIM card in slot 2 is afterwards not even listed in the menu. Yesterday, the phone said there was no data connection when I tried to browse the internet (SIM card 1) and after a restart it worked again.

**Comment by UniversalSuperBox** [comment]
Hi folks,

I believe this issue is similar to one that we've seen quite often with Ubuntu Touch. I've filed a new issue report that condenses the things we've learned from reports like this: https://github.com/ubports/ubuntu-touch/issues/1657

I believe this report can be closed in favor of that one, if you agree.
